---
float64[] viewpoints_x
float64[] viewpoints_y

